examples/flutter_gallery/lib/demo/material/chip_demo.dart

```diff
@@ -12,43 +76,241 @@ class ChipDemo extends StatefulWidget {
+          ),
+          label: new Text(_capitalize(name)),
+          onDeleted: () {
+            setState(() {
+              _removeTool(name);
+            });
+          });
+    }).toList();
+
+    final List<Widget> choiceChips = materials.map<Widget>((String name) {
+      return new ChoiceChip(
+        key: new ValueKey<String>(name),
+        backgroundColor: _nameToColor(name),
+        label: new Text(_capitalize(name)),
+        selected: selectedMaterial == name,
+        onSelected: (bool value) {
+          setState(() {
+            if (!value) {
+              selectedMaterial = '';
+            } else {
+              selectedMaterial = name;
+            }
+          });
+        },
+      );
+    }).toList();
 
-    if (_showBananas) {
-      chips.add(new Chip(
-        label: const Text('Bananas'),
-        onDeleted: _deleteBananas
-      ));
+    final List<Widget> filterChips = defaultTools.map<Widget>((String name) {
+      return new FilterChip(
+        key: new ValueKey<String>(name),
+        label: new Text(_capitalize(name)),
+        selected: tools.contains(name) ? selectedTools.contains(name) : false,
+        onSelected: tools.contains(name)
+            ? (bool value) {
+                setState(() {
+                  if (!value) {
+                    selectedTools.remove(name);
+                  } else {
+                    if (!selectedTools.contains(name)) {
+                      selectedTools.add(name);
+                    }
+                  }
+                });
+              }
+            : null,
+      );
+    }).toList();
+
+    Set<String> allowedActions = new Set<String>();
+    if (selectedMaterial != null && selectedMaterial.isNotEmpty) {
+      for (String tool in selectedTools) {
+        allowedActions = allowedActions.union(toolActions[tool]);
+      }
+      allowedActions = allowedActions.intersection(materialActions[selectedMaterial]);
     }
 
+    final List<Widget> actionChips = allowedActions.map<Widget>((String name) {
+      return new ActionChip(
+        label: new Text(_capitalize(name)),
+        onPressed: () {
+          setState(() {
+            selectedAction = name;
+          });
+        },
+      );
+    }).toList();
+
+    final ThemeData theme = Theme.of(context);
+    final List<Widget> tiles = <Widget>[
+      const SizedBox(height: 8.0, width: 0.0),
+      _wrapChips('Available Materials (Chip)', chips, theme),
+      _wrapChips('Available Tools (InputChip)', inputChips, theme),
+      _wrapChips('Choose a Material (ChoiceChip)', choiceChips, theme),
+      _wrapChips('Choose Tools (FilterChip)', filterChips, theme),
+      _wrapChips('Perform Allowed Action (ActionChip)', actionChips, theme),
+      const Divider(),
+      new Padding(
+        padding: const EdgeInsets.all(8.0),
+        child: new Center(
+          child: new Text(
+            _createResult(),
+            style: Theme.of(context).textTheme.title,
```

Comment:
You can use the `theme` var defined above.

Reply:
Done.